sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -325,6 +325,15 @@ public <T> Mono<T> findById(Object id, Class<T> domainType) {
     public <T> Mono<T> findById(String containerName, Object id, Class<T> domainType) {
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(domainType, "domainType should not be null");
+        CosmosContainerProperties containerProperties = getContainerProperties(containerName).block();
+        if (containerProperties != null) {
+            String containerPartitionKey = containerProperties.getPartitionKeyDefinition()
+                .getPaths().iterator().next().replaceAll("^/|/$", "");
+            if ("id".equals(containerPartitionKey)) {
+                return findById(id, domainType, new PartitionKey(CosmosUtils.getStringIDValue(id)));
+            }
+        }
+        LOGGER.warn("The partitionKey is not id!! Consider using findById(ID id, PartitionKey partitionKey) instead. See https://aka.ms/PointReadsInSpring for more info.");
         final String finalContainerName = getContainerNameOverride(containerName);
```

Comment:
I think this is causing some of your integration issues. We should call this above containerProperties and pass finalContainerName into getContainerProperties().

Reply:
Ditching getContainerProperties(), see above